pandas/tests/reshape/merge/test_merge.py

```diff
@@ -2096,6 +2096,23 @@ def test_merge_equal_cat_dtypes2():
     tm.assert_frame_equal(result, expected, check_categorical=False)
 
 
+def test_merge_on_cat_and_ext_array():
+    # GH 28668
+    df1 = DataFrame(
+        {"a": Series([pd.Interval(0, 1), pd.Interval(1, 2)], dtype="interval")}
+    )
+    df2 = df1.copy()
+    df2["a"] = df2["a"].astype("category")
+
+    joined1 = pd.merge(df1, df2, how="inner", on="a")
+    joined2 = pd.merge(df2, df1, how="inner", on="a")
+
```

Comment:
explicity construct the expected result rather than computing it

Reply:
Updated; to simplify the test a little I included only the merge that was actually erring on master (also allows to ignore the unrelated mismatching dtype issue)